### PR TITLE
Added location role

### PR DESCRIPTION
**Pull request by alawong**
Added a location role which loops through a list of locations foreman_locations and creates them on Satellite server using the foreman location module.

### CODE REVIEW

File: roles/locations/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+- name: Create locations  # noqa: args[module]
+  theforeman.foreman.location:
+    username: "{{ foreman_username | default(omit) }}"
+    password: "{{ foreman_password | default(omit) }}"
+    server_url: "{{ foreman_server_url | default(omit) }}"
+    validate_certs: "{{ foreman_validate_certs | default(omit) }}"
+    name: "{{ item.name }}"
+    parent: "{{ item.parent | default(omit) }}"
+    organizations: "{{ item.organizations | default(omit) }}"
+    parameters: "{{ item.parameters | default(omit) }}"
```
Comment (evgeni) [inline roles/locations/tasks/main.yml:11]:
```suggestion
    parameters: "{{ item.parameters | default(omit) }}"
    ignore_types: "{{ item.ignore_types | default(omit) }}"
```

you forgot this one parameter :)

File: roles/locations/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+theforeman.foreman.locations
+===================================
+
+This role creates and manages Satellite locations.
```
Comment (evgeni) [inline roles/locations/README.md]:
```suggestion
This role creates and manages Foreman locations.
```

(the branding happens automatically, here we keep the Foreman name)

**Review by evgeni — CHANGES_REQUESTED** [review]
Thanks Alan! And sorry it took so long to get this reviewed!

Overall, this looks good. I posted two comments inline, but nothing major.

One thing that is missing right now are tests. Given our test suite is a bit involved, I can understand if you say that you'd prefer not to touch even more things right now (and would offer you to just do the tests for you), or, if you want of course, you could try to write tests and I'll help you getting there? Absolutely up to you.

**Comment by alawong** [comment]
I have made the requested changes, and added the test playbook named locations_role.yml. Please review!

**Review by evgeni — APPROVED** [review]

**Pull request merged**